llvm/include/llvm/CodeGen/CodeGenPassBuilder.h

```diff
@@ -469,20 +486,36 @@ template <typename DerivedT> class CodeGenPassBuilder {
   const DerivedT &derived() const {
     return static_cast<const DerivedT &>(*this);
   }
+
+  void setStartStopPasses(const TargetPassConfig::StartStopInfo &Info) const;
+
+  Error verifyStartStop(const TargetPassConfig::StartStopInfo &Info) const;
+
+  mutable SmallVector<llvm::unique_function<bool(StringRef)>, 4>
```

Comment:
it seems nicer to keep this in `AddPass`. is the reason that these are moved out of `AddPass` because it references `Started`/`Stopped` in `CodeGenPassBuilder`? and there's no way to easily validate them if these are somewhere else?